### DISCUSSION TITLE
Meeting Notes

### DISCUSSION THREAD

**Discussion by MCooperBorkenhagen** [emelex-ai/BRIDGE#113] [General]
Agenda
--------
1. Check ins with students
2. Implementing a frequency-weighting more efficiently in simulations
3. Sequential machine teaching

Check ins with students
------------------------
Madhuri is working on a webpage for the project. Nathan will give her access to a Google machine to start work. The website will be relatively simple, but will include and interactive component (written in React) displaying the global embeddings.

Madhuri will be graduating/ leaving May 2025. Pramodh will be fall 2025.

Frequency weighting and sequential machine teaching
------------------------------------------------------

Our research questions require both curriculum learning and gradient weighting capabilities. We have determined that adding a frequency column to the data.csv file will support this. Here are the details:

#### Curriculum Learning
The original curriculum learning paradigm is possible with frequency weighting. Assume the data.csv file contains words duplicated by their frequency and we train the model on the words in the order they appear. We can set the frequency column to 1 for all words and every word's gradient will be weighted the same - effectively accomplishing curriculum learning.

#### Frequency Weighting
For pretraining or larger scale experiments duplicating words based on their frequencies can lead to impractically large files. Assume we want to train on 100,000 words including the word "the" and "aardvark". If we needed to duplicate the words based on their relative frequencies, then if "aardvark" appeared once in the dataset, then every other word would need to be duplicated in a propotional way. If "the" occurs 10,000 times more frequently than "aardvark" then it would need to be duplicated that many times, along with every other word. This would lead to an enormous data.csv file. Therefore, in this training scenario, we list every word a single time in the data.csv file and include non-uniform frequency weighting.

**Comment by mathnathan** [comment]
Great chat today!
